s32/drivers/s32ze/Dio/CMakeLists.txt

```diff
@@ -0,0 +1,5 @@
+# Copyright 2022 NXP
+# SPDX-License-Identifier: BSD-3-Clause
+
+zephyr_include_directories(include)
```

Comment:
Is this really needed?  (same question for any CMakeLists.txt) file.  I'd assume the header in here is local.

Reply:
these are the driver headers included in the shim drivers, e.g. https://github.com/zephyrproject-rtos/zephyr/pull/49161/files#diff-76940438b7185f8b5531917a8b22b2ae58b0f37ecbbf01c08ab31acc81c1babdR12-R13

Reply:
Can we simplify this to just be at the `drivers/s32ze` level so we can replace:

```
-I/Users/galak/git/modules/hal/nxp/s32/drivers/s32ze/BaseNXP/header 
-I/Users/galak/git/modules/hal/nxp/s32/drivers/s32ze/BaseNXP/include
-I/Users/galak/git/modules/hal/nxp/s32/drivers/s32ze/Dio/include
-I/Users/galak/git/modules/hal/nxp/s32/drivers/s32ze/Mcu/include
-I/Users/galak/git/modules/hal/nxp/s32/drivers/s32ze/Port/include 
-I/Users/galak/git/modules/hal/nxp/s32/drivers/s32ze/Rte/include 
-I/Users/galak/git/modules/hal/nxp/s32/drivers/s32ze/Uart/include 
```

with
```
-I/Users/galak/git/modules/hal/nxp/s32/drivers/s32ze
```

and tweak the `#include`s to be `#include <Port/include/Siul2_Port_Ip.h>`

Reply:
I agree this is a bit verbose. Although these headers are not only included by the shim drivers but also by the SDK baremetal drivers as well, so following suggested change implies tweaking all SDK driver includes as well. I prefer to avoid modifications to the SDK drivers unless necessary, to reduce maintenance effort when updating the SDK version or adding new baremetal drivers.